src/ChargeDrift/ChargeDrift.jl

```diff
@@ -287,20 +302,26 @@ function _drift_charge!(
     else
         zero(T)
     end
+    crystal_temperature::T = det.semiconductor.temperature
 
     last_real_step_index::Int = 1
     current_pos::Vector{CartesianPoint{T}} = deepcopy(startpos)
     step_vectors::Vector{CartesianVector{T}} = Vector{CartesianVector{T}}(undef, n_hits)
     done::Vector{Bool} = broadcast(pt -> !_is_next_point_in_det(pt, det, point_types), startpos)
     normal::Vector{Bool} = deepcopy(done)
+
+    # if the drifting model has a calculate_mobility function (scalar mobility), then we use the mobility-tied diffusion model
+    # TODO: adapt the code to 3D-vector mobility
+    use_mobility_tied_diffusion = hasproperty(det.semiconductor.charge_drift_model, :calculate_mobility)
     
     @inbounds for istep in 2:max_nsteps
         last_real_step_index += 1
         _set_to_zero_vector!(step_vectors)
-        _add_fieldvector_drift!(step_vectors, current_pos, done, electric_field, det, S)
+        _add_fieldvector_drift!(step_vectors, current_pos, done, electric_field, det, S, end_drift_when_no_field)
         self_repulsion && _add_fieldvector_selfrepulsion!(step_vectors, current_pos, done, charges, ϵ_r)
         _get_driftvectors!(step_vectors, done, Δt, det.semiconductor.charge_drift_model, current_pos, CC)
-        diffusion && _add_fieldvector_diffusion!(step_vectors, done, diffusion_length)
+        diffusion && !use_mobility_tied_diffusion && _add_fieldvector_diffusion!(step_vectors, done, diffusion_length)
+        diffusion && use_mobility_tied_diffusion && _add_fieldvector_diffusion_TiedWithMobility!(step_vectors, done, Δt, det.semiconductor.charge_drift_model.calculate_mobility, current_pos, crystal_temperature, CC)

```

Comment:
can you also combine these methods into one?
One could think about making `DiffusionModel` a struct that can become part of the `Semiconductor` or `ChargeDriftModel`.

Reply:
I have also thought about making a new struct for diffusion. But I am not sure about the goal. This depends on how we see "diffusion". Is it very clear physics process we should standardize? Or we provide options?

I would see "diffusion" and the Einstein's relation is quite clear. Maybe we should just standardize it and offer only one method `_add_fieldvector_diffusion_TiedWithMobility!`. This way we don't really need a struct. (We could upgrade it to 3D-vector diffusion length later)

Otherwise, we would need a struct, maybe it's better to build a new `ChargeDiffusionModel`

Reply:
That would also be an idea, but in general, the naming `_add_fieldvector_diffusion_TiedWithMobility!` is a bit weird...
Can you remove `_TiedWithMobility` from the name and then do something using multiple dispatch, for example:
```suggestion
        diffusion && (use_mobility_tied_diffusion ? _add_fieldvector_diffusion_TiedWithMobility!(step_vectors, done, Δt, det.semiconductor.charge_drift_model.calculate_mobility, current_pos, crystal_temperature, CC)  : _add_fieldvector_diffusion!(step_vectors, done, diffusion_length))
```